src/main/kotlin/FiToBookConvertor.kt

```diff
@@ -11,76 +11,112 @@ import java.time.format.DateTimeFormatter
-                dateFormat = dateFormatter,
-                startingRow = startingRow,
-                tradesSize = trades.size,
-                toClipboard = toClipboard,
-                summTabStartingRow = summRow,
-                summTabAmountCol = summAmountCol,
-                summTabBalCol = summBalCol
-            )
-        }
+        var row = startingRow + tradesSize - 1
+        var currentSummRow = summRow
+
+        activities.asReversed().forEach { activity ->
+            when (activity.type) {
+                in TRADE_TYPES -> append(formatSummaryTrade(row--, LocalDate.now().year))
+                else -> append(formatSummaryNonTrade(activity))
+            }
 
-        // Remove the last newline character
-        if (toClipboard.isNotEmpty()) {
-            toClipboard.deleteCharAt(toClipboard.length - 1)
+            if (summRow != null && summAmountCol != null && summBalCol != null) {
+                append(TAB).append("=$summBalCol$currentSummRow+$summAmountCol${++currentSummRow}")
+            }
+            append(NEWLINE)
         }
+    }
 
-        println("New clipboard:")
-        println(toClipboard)
-        writeToClipboard(toClipboard.toString())
+    private fun formatExecutionRow(activity: Activity, row: Int): String {
+        val correctedPrice = getCorrectedPrice(activity)
+        val amountFormula = when (activity.type) {
+            ActivityType.BOUGHT -> "=ROUND(-$SHARE_COL$row*$PRICE_COL$row-$FEE_COL$row,2)"
+            ActivityType.SOLD -> "=ROUND($SHARE_COL$row*$PRICE_COL$row-$FEE_COL$row,2)"
+            else -> throw IllegalStateException("Unsupported trade type: ${activity.type}")
+        }
 
-        findAndPrintDiscrepancy(trades, dateFormatter)
+        return buildString {
+            append(activity.date.format(dateFormatter)).append(TAB)
+            append(activity.shares).append(TAB)
+            append("${activity.type.sourceName}: ${activity.symbol}").append(TAB)
+            append(toAmount(correctedPrice)).append(TAB)
+            append(activity.fee!!.formattedAmount).append(TAB)
+            append(amountFormula)
+        }
     }
 
-    private fun findAndPrintDiscrepancy(trades: List<Activity>, dateFormat: DateTimeFormatter) {
-        val discrepancies = trades.mapNotNull { trade -> createDiscrepancyReport(trade, dateFormat) }
+    private fun formatSummaryTrade(row: Int, year: Int) = buildString {
+        append("='executions $year'!$DATE_COL$row").append(TAB)
+        append("=CONCATENATE(SUBSTITUTE('executions $year'!$OP_COL$row,\":\",CONCATENATE(\" \",'executions $year'!$SHARE_COL$row),1),\" @ \",TEXT('executions $year'!$PRICE_COL$row,\"$0.00\"))").append(TAB)
+        append("='executions $year'!$AMOUNT_COL$row")
+    }
 
-        if (discrepancies.isNotEmpty()) {
-            println("Discrepancies:")
-            discrepancies.forEach(::println)
-        }
+    private fun formatSummaryNonTrade(activity: Activity) = buildString {
```

Comment:
```suggestion
    private fun toInvestBookSummaryNonTradeActivity(activity: Activity) = buildString {
```
keep original method name